codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -224,6 +280,10 @@ private void decodeParams(String s) {
                     name = decodeComponent(s.substring(pos, i), charset);
                 }
                 pos = i + 1;
+            } else if (withCommaDecoding && c == ',' && name != null) {
```

Comment:
just wanted to make sure this behavior was intentional. the part that I'm not clear on is what you mentioned ... "reserved characters" are required to be percent encoded to interpret them as the literal value (https://www.ietf.org/rfc/rfc3986.txt section 2.2).